### DISCUSSION TITLE
no exports found in module

### DISCUSSION THREAD

**Discussion by zfreiter** [vercel/community#3690] [Help] [answered]
### Summary

I'm new to deploying a Node Express JS server and seem to be getting a, "No exports found in module "../server/index.js".
Did you forget to export a function or a server?" I deployed another Node Express JS server about 6 months ago and everything worked and that project still runs with no errors. The project runs fine locally with no errors. If anyone can help or point me in the right directions, it would be greatly appreciated.

### Steps to Reproduce

Below are the error messages, Vercel file, and the index file. 
![image](https://github.com/vercel/community/assets/31320978/8655b584-276a-447e-8f5b-c82a1f29937f)

No exports found in module "../server/index.js".
Did you forget to export a function or a server?
RequestId: dd60a738-0b68-4597-8b2c-de6b4c4abd9c Error: Runtime exited with error: exit status 1
Runtime.ExitError

Vercel file
![image](https://github.com/vercel/community/assets/31320978/dc52d961-2cd9-4e08-8378-81e3c831f327)

Index.js
![image](https://github.com/vercel/community/assets/31320978/d417889f-0ab5-46cf-ad7d-e557274d4432)

my json file 
![image](https://github.com/vercel/community/assets/31320978/b96c1c8a-3636-451f-a060-7a226736c20b)

**Comment by amyegan** [comment]
Hey @zfreiter. The [runtime logs](https://vercel.com/docs/observability/runtime-logs) usually have more error information. That's typically the first thing I check when I run into a [500: FUNCTION_INVOCATION_FAILED](https://vercel.com/docs/errors/application-error-codes#500:-function_invocation_failed) error in one of my deployments.

It may also be helpful to read through the guide on using Express.js with Vercel. It includes some [drawbacks and edge cases](https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases) to consider as well as recommendations about how to set up your Express site for serverless deployment. You can find the guide at https://vercel.com/guides/using-express-with-vercel

**Reply by zfreiter** [reply, marked as answer]
Thank you. I went through the logs and it provided some information about imports/libraries not being included. I ended up rewriting the connection to my mongodb with the mongoose library to a try catch.  I also retyped the body-parser import and everything started working.  I'm assuming it had an issue with the app.listen being nested in the then portion of the mongoose connection.
